### ISSUE TITLE
pittpatt "inconsistent image dimensions" errors on some devices

### ISSUE THREAD

**Issue by stevemadsenblippar** [googlesamples/android-vision#161]
On certain devices, face tracking fails with errors
`E/pittpatt: online_face_detector.cc:236] inconsistent image dimensions
E/pittpatt: detector.cc:220] inconsistent image dimensions`

Has been observed on
* Samsung Galaxy Note 4
* Samsung Galaxy J2
with apparently fairly normal front camera preview sizes of e.g.
960x540, 720x480, 640x480

One workaround is to force to 800x480, which enables face tracking to work
with front camera on both these devices (even though 800x480 not even listed
in available preview sizes on J2)

**Comment by pm0733464** [comment]
The face tracker assumes that each image frame received is consistently the same dimensions.  As far as I can tell, it looks like it is receiving a new image frame that is has different dimensions that the previous image frame.

Can you give more details on how you initialized the face detector and how you are passing in images?

**Comment by stevemadsenblippar** [comment]
Thank you for responding so quickly.

Our detector is very simple, just via 
`new FaceDetector.Builder().build()`
with 4 of the usual flags (`setLandmarkType(FaceDetector.NO_LANDMARKS)` etc)

Frame created via `new Frame.Builder().build()`, using `setImageData()` with a byteBuffer and our preview width and height (which will be constant until camera switched front/rear)

**Comment by pm0733464** [comment]
When the camera is switched or rotated, the face detector should be rebuilt so that it can reinitialize for images of a different dimension.

**Comment by stevemadsenblippar** [comment]
Yes -- the second I saw your original response I started looking at this, think the bug is in our code...mea culpa
